### PR TITLE
New coords handler updated to use xyz coordinates

### PR DESCRIPTION
**Pull request by saramazz**
The functions `run_bayesian_analysis_router`, `area`, and `coordinates` have been modified to utilize `dt_papers_nq` and `xyz_coords`, which are calculated by the `load_or_calculate_variables` function from the `data-neurosynth_version-7_coordinates.tsv` file and affine matrices. In the case of `area`, it is `mask, affine = generate_nifit_mask(area, atlas_path)`.

For `coordinates`, it is `mask, affine = generate_nifit_mask(1, atlas_path)`. 

The time taken to obtain the results has been significantly reduced.

### CODE REVIEW

File: neuroinfer/code/BayesianAnalysis.py [hunk 1]
```diff
@@ -24,19 +29,33 @@ def run_bayesian_analysis_router(cog_list, area, prior_list, x_target, y_target,
...
 
+    # Load mask image   
+    #mask_nifti = image.load_img('.tmp/mask.nii.gz')
+    #affine_inv = np.linalg.inv(mask_nifti.affine) #inverse of the affine matrix to convert from MNI coordinates tp voxel coordinates
+    
 
     if not pd.isnull(area) and pd.isnull(x_target) and pd.isnull(y_target) and pd.isnull(z_target):
         # Call run_bayesian_analysis_area if area is not nan and coordinates are nan
-        results=run_bayesian_analysis_area(cog_list, prior_list, area, radius, result_df, cm)
+        #results=run_bayesian_analysis_area(cog_list, prior_list, area, radius, result_df, cm)
```
Comment (mlionello) [inline neuroinfer/code/BayesianAnalysis.py]:
```suggestion
```

File: neuroinfer/code/BayesianAnalysis.py [hunk 2]
```diff
@@ -47,7 +66,22 @@ def run_bayesian_analysis_router(cog_list, area, prior_list, x_target, y_target,
 
     elif not np.isnan(x_target) and not np.isnan(y_target) and not np.isnan(z_target) and np.isnan(area):
         # Call run_bayesian_analysis_coordinates if coordinates are not nan and area is nan
-        results=run_bayesian_analysis_coordinates(cog_list, prior_list, x_target, y_target, z_target, radius, result_df, cm)
+        mask,affine=generate_nifit_mask(1,atlas_path) #area is set to 1 to get the mask at the coordinates
+        affine_inv = np.linalg.inv(affine) #inverse of the affine matrix to convert from MNI coordinates tp voxel coordinates
+
+        
+
+        #dt_papers_nq = pd.read_csv(os.path.join(data_path, 'data-neurosynth_version-7_coordinates.tsv'), sep='\t')
```
Comment (mlionello) [inline neuroinfer/code/BayesianAnalysis.py]:
```suggestion
```

File: neuroinfer/code/BayesianAnalysis.py [hunk 3]
```diff
@@ -24,19 +29,33 @@ def run_bayesian_analysis_router(cog_list, area, prior_list, x_target, y_target,
...
+    data_path = os.path.join(global_path, "data")  # Path to the saved_result folder
+    #print('Data path: ',data_path)
 
     # Check if results_folder_path exists, if not, create it
     if not os.path.exists(results_folder_path):
         os.makedirs(results_folder_path)
 
+    # Load mask image   
+    #mask_nifti = image.load_img('.tmp/mask.nii.gz')
+    #affine_inv = np.linalg.inv(mask_nifti.affine) #inverse of the affine matrix to convert from MNI coordinates tp voxel coordinates
```
Comment (mlionello) [inline neuroinfer/code/BayesianAnalysis.py]:
```suggestion
```

**Review by mlionello — CHANGES_REQUESTED** [review]
remove comments

**Review by mlionello — APPROVED** [review]

**Review by davidecrc — APPROVED** [review]

**Pull request merged**